va_explorer/va_data_management/views.py

```diff
@@ -1,18 +1,36 @@
 from django.shortcuts import render, redirect
 from .models import VerbalAutopsy, CauseOfDeath, CauseCodingIssue, Location
 from .forms import VerbalAutopsyForm
+from django.core.paginator import Paginator, PageNotAnInteger, EmptyPage
 
-def index(request):
-    va_list = VerbalAutopsy.objects.prefetch_related("location", "causes", "coding_issues").order_by("id")
-    va_data = [{
+def index(request, page_size=15):
+    # active page - default to 1 if not active page
+    absolute_page = int(request.GET.get('page', 1))
+
+    # query to retrive all VAs
+    va_list = (VerbalAutopsy.objects
+        .prefetch_related("location", "causes", "coding_issues")
+        .order_by("id")
+    )
+
+    # create paginator from query and bucket vas into pages of size page_size
+    paginator = Paginator(va_list, page_size)
+
+    # hone in on active page objects
+    current_page = paginator.page(absolute_page)
```

Comment:
We can just call `get_page` instead of `page` then we don't need to do the conversion to `int()` manually above. It's a little safer 

https://docs.djangoproject.com/en/3.1/ref/paginator/#django.core.paginator.Paginator.get_page

Reply:
Thanks for pointing this out. Will change to get_page